### DISCUSSION TITLE
Access to binary representation of embedded messages.

### DISCUSSION THREAD

**Discussion by FSMaxB** [protobuf-c/protobuf-c#647] [Q&A]
I have a message `A` that embeds another message `B`. When I unpack `A`, `B` gets automatically unpacked as well.

What I need is access to the binary data of `B` in order to authenticate it using an AEAD scheme (authenticated encryption with additional data). I thought about taking the unpacked `B` and packing it again, but I don't think there is a guarantee that it will be exactly the same as the original packed `B`.

Is there a way that I can get access to the binary representation of `B`?

**Comment by lipnitsk** [comment]
Don't think protobuf-c can do this. The best way I can think of would be to declare `B` as `bytes` in the `.proto` file, then unpack it explicitly.
